src/components/PopupWithForm.js

```diff
@@ -17,15 +18,38 @@ export default class PopupWithForm extends Popup {
   }
 
   reset() {
-    this._popupForm.reset();
+    if (this._popupForm) {
+      this._popupForm.reset();
+    }
   }
 
   setEventListeners() {
-    this._popupForm.addEventListener("submit", (e) => {
-      e.preventDefault();
-      this._handleFormSubmit(this._getInputValues());
-    });
-
     super.setEventListeners();
+    if (this._popupForm) {
+      this._popupForm.addEventListener("submit", (e) => {
+        e.preventDefault();
+        this._handleFormSubmit(this._getInputValues());
+      });
+    }
+  }
+
+  open() {
+    super.open();
+  }
+
+  setSubmitHandler(newSubmitHandler) {
```

Comment:
Needs correction: you should create a separate class for handling the popup with confirmation `PopupWithConfirmation`